src/MOI_wrapper.jl

```diff
@@ -0,0 +1,355 @@
+using LinearAlgebra
+import MathOptInterface as MOI
+using ManifoldsBase
+using ManifoldDiff
+
+include("qp_block_data.jl")
+
+const _FUNCTIONS = Union{
+    MOI.VariableIndex,MOI.ScalarAffineFunction{Float64},MOI.ScalarQuadraticFunction{Float64}
+}
+
+"""
+    struct VectorizedManifold{M} <: MOI.AbstractVectorSet
+        manifold::M
+    end
+
+Representation of points of `manifold` as a vector of `R^n` where `n` is
+`MOI.dimension(VectorizedManifold(manifold))`.
+"""
+struct VectorizedManifold{M} <: MOI.AbstractVectorSet
+    manifold::M
+end
+
+function MOI.dimension(set::VectorizedManifold)
```

Comment:
Should this maybe be documented? Then we could do a small `extensions/moi.md` documentation page for these functions here. I anyways wanted to start those pages to extensions that we do here.

Reply:
Done, let's maybe create this documentation page as a follow up PR ?